### ISSUE TITLE
{New Feature] add nuker to auto mine any type of block nearby

### ISSUE THREAD

**Issue by Shaurya4356** [MCCTeam/Minecraft-Console-Client#2017]
### Prerequisites

- [X] I have read and understood the [user manual](https://github.com/MCCTeam/Minecraft-Console-Client/tree/master/MinecraftClient/config)
- [X] I made sure I am running the latest [development build](https://ci.appveyor.com/project/ORelio/Minecraft-Console-Client/build/artifacts)
- [X] I tried to [look for similar feature requests](https://github.com/MCCTeam/Minecraft-Console-Client/issues?q=is%3Aissue) before opening a new one

### Console Client Version

latest

### Describe your problem

It will be super good if alt an nuke any block nearby (in 4 block radius)
, make sure to make it work even while alt is in minecart or riding entities

### Suggest a possible solution

make cobblestone miner into universal nuker or just remove x y z location and make it mine anything in range

### Attach screenshot here (If applicable)

d

### Minecraft Version

1.16.5

### Device

Desktop

### Operating System

Windows

**Comment by Daenges** [comment]
While we already have [MineCube.cs](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/config/ChatBots/MineCube.cs), which operates like the `/fill` command, I do not see a nuker (a feature mostly known from modified clients to grief other players property) in MCC, for the same reason a killaura was declined. From a technical perspective this would be no problem. Due to the fact that MCC is open source nobody can prevent you from creating a fork and implement such a feature on your own.

Although it is not meant to, MCC provides (for obvious reasons) many possibilities to mess around on servers that are not your own. I do not see any another justification for a Nuker other than griefing other players (if you just want to mine something out, use MineCube.cs, which is currently even reworked #2014 ).

So, to summarize: MCC is supposed to be a helpful tool for admins or on servers which allow the usage. We discourage cheating with MCC and your suggestion would promote it.

If you could provide a use case of this, other than griefing/cheating, we can talk about it again.

**Comment by Shaurya4356** [comment]
https://github.com/MCCTeam/Minecraft-Console-Client/issues/2017#issuecomment-1122695811

no no, i dont want it for some griefing purpose lmao
it happens to be , that my server has realistic tree falll animation and auto sapling placement plugin
and i want the nuker just for my minecart-auto tree machine

becuase unlike others i dont have strong pc to run multiple alts- i think this is the objective of mcc client

**Comment by Daenges** [comment]
Although you might not use it for malicious purposes, adding this feature for everyone sounds dangerous.

If you are using a minecart, you could use use a slightly altered version of the [cobblestoneminer.cs](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/config/ChatBots/CobblestoneMiner.cs). What you described as `select a cube and mine everything in it` is what [MineCube.cs](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/config/ChatBots/MineCube.cs) does. There you would not even need a minecart, since the client automatically walks to blocks it is going to mine.

If you want to mine a specific cube over and over again, just use a task for it.

Only disadvantage is, that the script is currently in a beta state and might break at some points. But therefore I am currently reworking it.

**Comment by Shaurya4356** [comment]
https://github.com/MCCTeam/Minecraft-Console-Client/issues/2017#issuecomment-1134392795
 
 fare enough, your concern is also crucial
 
 
 how about add a autoclicker, with 4
 modes=hold/click
 button= left/right
 delay in case of mode=click
 
 
 it will do the job

**Comment by Daenges** [comment]
If I understood your problem correctly, you want to use the client to break blocks.
You can edit the [cobblestone miner](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/config/ChatBots/CobblestoneMiner.cs) to break any block at the selected location by removing the switch statement:

<details>
<summary>Like this:  (Click me)</summary>

```cs

//MCCScript 1.0

MCC.LoadBot(new CobblestoneMiner());

//MCCScript Extensions

public class CobblestoneMiner: ChatBot
{
    // === CONFIG - REPLACE COBBLESTONE LOCATION x y z VALUES HERE ===
    // You need to stand in front of the cobblestone block to mine
    // Also make sure the Cobblestone will regenerate e.g. using water and lava
    Location cobblestone = new Location(x, y, z);
    // === END OF CONFIG ===

    public override void Initialize()
    {
        LogToConsole("Bot enabled!");
    }

    public override void Update()
    {
        DigBlock(cobblestone);
    }
}
```


</details>

Or if you need a relative position, since you are constantly moving in a minecart..

<details>
<summary>Like this:</summary>

```cs
//MCCScript 1.0

MCC.LoadBot(new CobblestoneMiner());

//MCCScript Extensions

public class CobblestoneMiner: ChatBot
{
    Location relativeLocationToMine = Location.Zero;

    public override void Initialize()
    {
        LogToConsole("Bot enabled!");
    }

    public override void Update()
    {
        relativeLocationToMine = GetCurrentLocation() + new Location(/*RelativeX*/, /*RelativeY*/, /*RelativeZ*/);
        DigBlock(relativeLocationToMine);
    }
}
```

</details>

This should be able to dig a block on your desired position. Here is my [tutorial](https://youtu.be/JbDpwwETEnU?t=347) on how to set up this script. Did not test the code, so let me know if you got errors.

**Comment by Shaurya4356** [comment]
i want relative position- yes
if possible make it block selective, cuz i dont want to break saplings
and how to add multiple relative positions?

**Comment by Shaurya4356** [comment]
//MCCScript 1.0

MCC.LoadBot(new CobblestoneMiner());

//MCCScript Extensions

public class CobblestoneMiner: ChatBot
{
    Location relativeLocationToMine = Location.Zero;

    public override void Initialize()
    {
        LogToConsole("Bot enabled!");
    }

    public override void Update()
    {
        relativeLocationToMine = GetCurrentLocation() + new Location(/*RelativeX*/, /*RelativeY*/, /*RelativeZ*/);
        Material blockType = GetWorld().GetBlock(relativeLocationToMine).Type;
        switch (blockType)
        {
            case Material.Stone:
                DigBlock(relativeLocationToMine);
                break;
            case Material.Cobblestone:
                DigBlock(relativeLocationToMine);
                break;
        }
    }
}

**Comment by Shaurya4356** [comment]
and how to encode spruce log into material, the code for it?

**Comment by Daenges** [comment]
(Material.SpruceLog)[https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/Mapping/Material.cs#L53]

**Comment by Daenges** [comment]
> and how to add multiple relative positions?
In the current state of the script your can not do this. Just place the rails in a way that gets the blocks in the relative spot that is mined anyway.

**Comment by Shaurya4356** [comment]
and yes how to correcting format this 
 relativeLocationToMine = GetCurrentLocation() + new Location(/*RelativeX*/, /*RelativeY*/, /*RelativeZ*/);
if i want x+1, y-1, z+2

**Comment by Daenges** [comment]
`new Location(1, -1, 2);`

**Comment by Shaurya4356** [comment]
//MCCScript 1.0

MCC.LoadBot(new CobblestoneMiner());

//MCCScript Extensions

public class CobblestoneMiner: ChatBot
{
    Location relativeLocationToMine = Location.Zero;

    public override void Initialize()
    {
        LogToConsole("Bot enabled!");
    }

    public override void Update()
    {
        relativeLocationToMine = GetCurrentLocation() + new Location(+1,+1,0);
        Material blockType = GetWorld().GetBlock(relativeLocationToMine).Type;
        switch (blockType)
        {
            case Material.SpruceLog:
                DigBlock(relativeLocationToMine);
                break;
            
        }
    }
}

this generate this error
[WARN] Update: Got error from ScriptLoader.Script+CobblestoneMiner: System.NullReferenceException: Object reference not set to an instance of an object.
   at ScriptLoader.Script.CobblestoneMiner.Update()
   at MinecraftClient.McClient.OnUpdate()

**Comment by Shaurya4356** [comment]
`//MCCScript 1.0

MCC.LoadBot(new CobblestoneMiner());

//MCCScript Extensions

public class CobblestoneMiner: ChatBot
{
Location relativeLocationToMine = Location.Zero;

public override void Initialize()
{
    LogToConsole("Bot enabled!");
}

public override void Update()
{
    relativeLocationToMine = GetCurrentLocation() + new Location(+1,+1,0);
    Material blockType = GetWorld().GetBlock(relativeLocationToMine).Type;
    switch (blockType)
    {
        case Material.SpruceLog:
            DigBlock(relativeLocationToMine);
            break;
        
    }
}
}`

this generate this error
[WARN] Update: Got error from ScriptLoader.Script+CobblestoneMiner: System.NullReferenceException: Object reference not set to an instance of an object.
at ScriptLoader.Script.CobblestoneMiner.Update()
at MinecraftClient.McClient.OnUpdate()

**Comment by Daenges** [comment]
Okay. After searching what the problem might be for way too long... You need to turn on [terrainandmovements](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/Resources/config/MinecraftClient.ini#L35) and inventoryhandling in the config.

**Comment by Shaurya4356** [comment]
works, if player is on ground, but i doesn't while player is in minecart
it even does not swing his axe in minecart

**Comment by Daenges** [comment]
If the minecart version does not work, you need to rely on walking. The [MineCube](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/config/ChatBots/MineCube.cs) script should work here as long as there is a 2 high area free from blocks above the cube you want to mine. It will try to mine all blocks in a selected area.

**Comment by Daenges** [comment]
You can also get the WIP version [here](https://github.com/MCCTeam/Minecraft-Console-Client/blob/1eee1b16fa1fa3233caf2d0447c1f2161e6db91e/MinecraftClient/config/ChatBots/MineCube.cs).
You could stop the time, how long the script needs to clear the area and write a task to start it from time to time.

**Comment by milutinke** [comment]
We won't add any features that could be used for griefing to the client.
